### PR TITLE
Estimator to return 1 when there are no leases

### PR DESCRIPTION
**Pull request by ealsur**
This PR will make the `GetEstimatedRemainingWork` return 1 instead of 0 if there are no leases initialized.

### CODE REVIEW

File: src/DocumentDB.ChangeFeedProcessor/PartitionManagement/RemainingWorkEstimator.cs [hunk 1]
```diff
@@ -44,6 +45,7 @@ public RemainingWorkEstimator(ILeaseManager leaseManager, IChangeFeedDocumentCli
 
             foreach (ILease existingLease in await this.leaseManager.ListAllLeasesAsync().ConfigureAwait(false))
             {
+                hasAtLeastOneLease = true;
```
[the remark this answers is not in the document]
Comment (ealsur) [reply src/DocumentDB.ChangeFeedProcessor/PartitionManagement/RemainingWorkEstimator.cs]: Done

File: src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs [hunk 2]
```diff
@@ -126,5 +126,72 @@ public EstimatorTests(IntegrationTestFixture fixture) : base(fixture, typeof(Est
...
+                this.LeaseCollectionInfo,
+                new ChangeFeedOptions { StartFromBeginning = false },
+                new ChangeFeedHostOptions());
+
+            // Initialize leases
+            await host.RegisterObserverFactoryAsync(observerFactory);
+            // Stop host
+            await host.UnregisterObserversAsync();
+
+            using (var client = new DocumentClient(this.ClassData.monitoredCollectionInfo.Uri, this.ClassData.monitoredCollectionInfo.MasterKey, this.ClassData.monitoredCollectionInfo.ConnectionPolicy))
```
Comment (mkolt) [inline src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs]: the line is too long

File: src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs [hunk 3]
```diff
@@ -126,5 +126,72 @@ public EstimatorTests(IntegrationTestFixture fixture) : base(fixture, typeof(Est
...
+            var hostName = Guid.NewGuid().ToString();
+
+            var host = new ChangeFeedEventHost(
+                hostName,
+                this.ClassData.monitoredCollectionInfo,
+                this.LeaseCollectionInfo,
+                new ChangeFeedOptions { StartFromBeginning = false },
+                new ChangeFeedHostOptions());
+
+            // Initialize leases
```
Comment (mkolt) [inline src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs]: Please clarify what this step is doing and why it's needed.

File: src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs [hunk 4]
```diff
@@ -126,5 +126,72 @@ public EstimatorTests(IntegrationTestFixture fixture) : base(fixture, typeof(Est
...
+            await host.RegisterObserverFactoryAsync(observerFactory);
+            // Stop host
+            await host.UnregisterObserversAsync();
+
+            using (var client = new DocumentClient(this.ClassData.monitoredCollectionInfo.Uri, this.ClassData.monitoredCollectionInfo.MasterKey, this.ClassData.monitoredCollectionInfo.ConnectionPolicy))
+            {
+                await IntegrationTestsHelper.CreateDocumentsAsync(
```
Comment (mkolt) [inline src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs]: why do we need to create documents?

Comment (ealsur) [reply src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs]: This is to validate that, when leases have no ContinuationToken, we are using StartFromBeginning. With that, we can count the inserted documents.

File: src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs (hunk 3 above)
Comment (ealsur) [reply src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs]: I will add comments. This will initialize the host and generate leases. The leases will have ContinuationToken null initially.

**Comment by ealsur** [comment]
Added comments in code.
Basically the tests cover 2 scenarios:
* There are no leases, in which case, it returns 1
* The leases have no ContinuationToken, in which case, we use StartFromBeginning

File: src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs (hunk 4 above)
Comment (mkolt) [reply src/DocumentDB.ChangeFeedProcessor.IntegrationTests/EstimatorTests.cs]: I see. Can we also have a test for the case when lease is there with null continuation token, and validate that we return 1 for remaining work?
